pulp_smash/tests/puppet/cli/test_sync_from_puppet_skip_duplicate_content.py

```diff
@@ -0,0 +1,70 @@
+# coding=utf-8
+"""Tests that to verify Puppet sync on duplicate contents."""
+from __future__ import unicode_literals
+
+import unittest2
+
+from pulp_smash import cli, config, utils
+from pulp_smash.constants import PUPPET_FEED, PUPPET_QUERY
+from pulp_smash.tests.puppet.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+
+class SyncFromPuppetSkipDuplicateContentTestCase(unittest2.TestCase):
+    """Test whether the importer recognizes the content already in Pulp.
+
+    According to `Pulp #1937`_, Pulp sometimes fails to sync units into a
+    repository if the unit already existed.
+
+    How do we test this? The test steps would be as follows:
+
+    1. upload or sync a module to repo1
+    2. perform a sync on repo2 against a forge-style repository
+        (one with a modules.json file) that includes the module from 1.
+    3. Assert the module from 1. is in the published repo2.
+        (i.e., just check if the 2nd sync would succeed)
+    Please see reference from: https://github.com/PulpQE/pulp-smash/issues/269
+
+    To do this in pulp-admin, it could be something as simple as:
+
+    1. pulp-admin puppet repo create --repo-id=repo1
+        --feed=http://forge.puppetlabs.com --queries=torssh
+    2. pulp-admin puppet repo sync run --repo-id=repo1
+    3. pulp-admin puppet repo create --repo-id=repo2
+        --feed=http://forge.puppetlabs.com --queries=torssh
+    4. pulp-admin puppet repo sync run --repo-id=repo2
```

Comment:
Heh. I appreciate thorough documentation. Even so, this is a bit verbose. We want to tell *what*, not *how*. We want to be able to do things like change the `--queries=torssh` in our test without having to also update this docstring.